### PR TITLE
Add hocon filetype

### PR DESCRIPTION
**Pull request by GabrielNagy**
Add support for commenting hocon files.

[1] https://github.com/lightbend/config/blob/main/HOCON.md#comments

### CODE REVIEW

**Comment by alerque** [comment]
I don't know anything about hocon, but the docs you link to are ... misinformed.

> As in standard JSON and standard HOCON, any text between // or # and the next newline is considered a comment and is ignored (unless the // or # is inside a quoted string).

Standard JSON does not have a valid comment syntax at all. This is something various derivatives frequently change, but it's not encouraging that their documentation doesn't know where they are starting from.

File: plugin/NERD_commenter.vim [hunk 1]
```diff
@@ -204,6 +204,7 @@ let s:delimiterMap = {
     \ 'hbs': { 'left': '{{!-- ', 'right': ' --}}' },
     \ 'hercules': { 'left': '//', 'leftAlt': '/*', 'rightAlt': '*/' },
     \ 'hive': { 'left': '-- ' },
+    \ 'hocon': { 'left': '#' },
```
Comment (alerque) [inline plugin/NERD_commenter.vim:207]: Why use `#` as the default? The docs you linked to used `//` in most of their examples, it seems like that might be more idiomatic. Whichever one you use, you should also include a `leftAlt` key with the other one.

**Review by alerque — CHANGES_REQUESTED** [review]
Thanks for taking the time to contribute.

**Comment by GabrielNagy** [comment]
> Standard JSON does not have a valid comment syntax at all. This is something various derivatives frequently change, but it's not encouraging that their documentation doesn't know where they are starting from.

I agree, I just clicked on the first google result. I think this is the correct source: https://github.com/lightbend/config/blob/main/HOCON.md#comments

I'll update the code.

File: plugin/NERD_commenter.vim (hunk 1 above)
Comment (GabrielNagy) [reply plugin/NERD_commenter.vim]: No reason in particular, just Ruby mnemonics. I updated the code to default on `//`.

**Review by alerque — APPROVED** [review]

**Pull request merged**